Libraries/LLM/LLMModel.swift

```diff
@@ -50,6 +50,23 @@ public actor ModelContainer {
             tokenizerConfig: tokenizerConfig, tokenizerData: tokenizerData)
     }
 
+    public func numParams() -> Int {
+        return model.leafModules().flattenedValues().map {
+            mod -> Int in
+            if let qlin = mod as? QuantizedLinear {
+                return qlin.scales.size * qlin.groupSize
+            } else if let qemb = mod as? QuantizedEmbedding {
+                return qemb.scales.size * qemb.groupSize
```

Comment:
We could add a `Quantized` protocol so that all layers current/future would work with this. Something like:

```swift
var numParams: Int { get }
```

then each type would implement that as needed (encapsulation).  That would require a change on the mlx-swift side, but might be the way to go.  I was thinking we might need something like this anyway as currently QuantizedLinear is a subtype of Linear and thus ... I think you can quantize a QuantizedLinear.  Ugh.